opium_kernel/middlewares/logger.ml

```diff
@@ -0,0 +1,66 @@
+(** [Logger] logs the requests and their response.
+
+    The request's target URL and the HTTP method are logged with the "info" verbosity.
+    Once the request has been processed successfully, the response's HTTP code is logged
+    with the "info" verbosity.
+
+    If the body of the request or the response are a string (as opposed to a stream),
+    their content is logged with the "debug" verbosity.
+
+    If an error occurs while processing the request, the error is logged with an "error"
+    verbosity.
+
+    Note that this middleware is best used as the first middleware of the pipeline because
+    any previous middleware might change the request / response after [Logger] has been
+    applied. *)
+
+open Core
+open Core.Rock
+
+let log_src = Logs.Src.create "opium.server"
+
+let log_body Core.Body.{ content; _ } =
+  match content with
+  | `Stream _ -> ()
+  | `Empty -> ()
+  | `String s -> Logs.debug ~src:log_src (fun m -> m "%s" s)
+  | `Bigstring b -> Logs.debug ~src:log_src (fun m -> m "%s" (Bigstringaf.to_string b))
+;;
```

Comment:
This could be simplified by using the existing pretty printers for the Body type

```suggestion
let log_body body = Logs.debug ~src:log_src (fun m -> m "%a" Core.Body.pp_hum body)
```